### ISSUE TITLE
Feature Request: Perform app-local dll copy also based on the downstream library instead just downstream application

### ISSUE THREAD

**Issue by jasjuang** [Microsoft/vcpkg#2681]
Can we add a functionality so the app-local dll copy can be performed also based on the downstream library instead of just the downstream application? Below is a minimal example to illustrate what I am talking about.

CMakeLists
```
cmake_minimum_required(VERSION 3.9)

project(Example)

set(PROJECT_SRCS ${PROJECT_SOURCE_DIR}/mycpp.cpp) 

find_package( tinyxml2 REQUIRED )

add_library(${PROJECT_NAME} SHARED ${PROJECT_SRCS})

target_link_libraries(${PROJECT_NAME} tinyxml2)
```

mycpp.cpp
```
#include <tinyxml2.h>
int main(){
  tinyxml2::XMLDocument xml;
  return 0;
}
```

After building it `tinyxml2.dll` will not be copied to where `Example.dll` is. Despite running dumpbin on `Example.dll` shows it depends on `tinyxml2.dll`. However if I change my CMakeLists to 
```
cmake_minimum_required(VERSION 3.9)

project(Example)

set(PROJECT_SRCS ${PROJECT_SOURCE_DIR}/mycpp.cpp) 

find_package( tinyxml2 REQUIRED )

add_executable(${PROJECT_NAME} ${PROJECT_SRCS})

target_link_libraries(${PROJECT_NAME} tinyxml2)
```

it does got copied.

**Comment by jasjuang** [comment]
@ras0219-msft Could you please look into this? My understanding in powershell is really limited so I am having a hard time to figure this out myself. I am not sure what to change in `applocal.ps1` so the dumpbin will be invoked not only on user generated EXE but also DLL as well.

**Comment by ras0219-msft** [comment]
Thanks for the suggestion!

I've pushed `applocal-library-deps` to try this out; you'll need to `set(VCPKG_APPLOCAL_LIBRARY_DEPS ON)` to enable it.

Let me know if that solves the issue!

**Comment by jasjuang** [comment]
@ras0219-msft I tested it out and it works. Thanks for implementing it. However, shouldn't this by default be on? Why do we need an additional macro to enable this?

**Comment by brucedjones** [comment]
Coincidentally I ran into this problem a few hours ago and implemented a fix (#2787) along very similar lines before finding this thread. Thanks @jasjuang for making the connection.

I'm inclined to agree with @jasjuang  that I don't think we need a separate flag for this. Coming in to this as a new user I simply expected applocal_deps to apply to both executables and libraries.

**Comment by jasjuang** [comment]
Fixed by #2787
